modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -1577,7 +1583,23 @@ cv::gimpl::ov::GOVExecutable::GOVExecutable(const ade::Graph &g,
         }
     }
 }
+#if defined HAVE_INF_ENGINE && INF_ENGINE_RELEASE >= 2024030000
+cv::gimpl::ov::GOVExecutable::~GOVExecutable() {
+    if(m_workload.has_value())
+        m_workload.value().get().removeListener(m_workloadId);
+}
 
+void cv::gimpl::ov::GOVExecutable::setWorkLoadType(const std::string &type) {
```

Comment:
Why for OpenVINO backend we place the implementation into the `GOVExecutable`, while for ONNX we place it in the `ONNXCompiled`?